src/diffusers/pipelines/auto_pipeline.py

```diff
@@ -0,0 +1,142 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+
+from ..configuration_utils import ConfigMixin
+from .deepfloyd_if import IFImg2ImgPipeline, IFInpaintingPipeline, IFPipeline
+from .kandinsky import KandinskyImg2ImgPipeline, KandinskyInpaintPipeline, KandinskyPipeline
+from .kandinsky2_2 import KandinskyV22Img2ImgPipeline, KandinskyV22InpaintPipeline, KandinskyV22Pipeline
+from .stable_diffusion import (
+    StableDiffusionImg2ImgPipeline,
+    StableDiffusionInpaintPipeline,
+    StableDiffusionPipeline,
+)
+from .stable_diffusion_xl import (
+    StableDiffusionXLImg2ImgPipeline,
+    StableDiffusionXLInpaintPipeline,
+    StableDiffusionXLPipeline,
+)
+
+
+AUTO_TEXT2IMAGE_PIPELINES_MAPPING = OrderedDict(
+    [
+        ("stable-diffusion", StableDiffusionPipeline),
+        ("stable-diffusion-xl", StableDiffusionXLPipeline),
+        ("if", IFPipeline),
+        ("kandinsky", KandinskyPipeline),
+        ("kdnsinskyv22", KandinskyV22Pipeline),
+    ]
+)
+
+AUTO_IMAGE2IMAGE_PIPELINES_MAPPING = OrderedDict(
+    [
+        ("stable-diffusion", StableDiffusionImg2ImgPipeline),
+        ("stable-diffusion-xl", StableDiffusionXLImg2ImgPipeline),
+        ("if", IFImg2ImgPipeline),
+        ("kandinsky", KandinskyImg2ImgPipeline),
+        ("kdnsinskyv22", KandinskyV22Img2ImgPipeline),
+    ]
+)
+
+AUTO_INPAINTING_PIPELINES_MAPPING = OrderedDict(
+    [
+        ("stable-diffusion", StableDiffusionInpaintPipeline),
+        ("stable-diffusion-xl", StableDiffusionXLInpaintPipeline),
+        ("if", IFInpaintingPipeline),
+        ("kandinsky", KandinskyInpaintPipeline),
+        ("kdnsinskyv22", KandinskyV22InpaintPipeline),
+    ]
+)
+
+SUPPORTED_TASKS_MAPPINGS = [
+    AUTO_TEXT2IMAGE_PIPELINES_MAPPING,
+    AUTO_IMAGE2IMAGE_PIPELINES_MAPPING,
+    AUTO_INPAINTING_PIPELINES_MAPPING,
+]
+
+
+def _get_task_class(mapping, pipeline_class_name):
+    def get_model(pipeline_class_name):
+        for task_mapping in SUPPORTED_TASKS_MAPPINGS:
+            for model_name, pipeline in task_mapping.items():
+                if pipeline.__name__ == pipeline_class_name:
+                    return model_name
+
+    model_name = get_model(pipeline_class_name)
+
+    if model_name is not None:
+        task_class = mapping.get(model_name, None)
+        if task_class is not None:
+            return task_class
+    raise ValueError(
+        f"AutoPipeline can't find a pipeline linked to {pipeline_class_name} for {model_name} in {mapping}"
+    )
+
+
+class AutoPipelineForText2Image(ConfigMixin):
+    config_name = "model_index.json"
+
+    @classmethod
+    def from_pretrained(cls, pretrained_model_or_path, **kwargs):
+        config = cls.load_config(pretrained_model_or_path)
+
+        text_2_image_cls = _get_task_class(AUTO_TEXT2IMAGE_PIPELINES_MAPPING, config["_class_name"])
+
+        return text_2_image_cls.from_pretrained(pretrained_model_or_path, **kwargs)
```

Comment:
Here we actually need to do a bit more because `.components()` doesn't have all the necessary inputs necessarily. E.g. SD has a `require_safety_checker` pipeline optional attribute that we need to pass here as well. What you could /should do here therefore is the following:

1. Retrieve all **optional kwargs** that the pipeline can accept. You can have a look here at how this can be done:
https://github.com/huggingface/diffusers/blob/c6e56e92ed105033e3bfd66604ef3ebe2646c26c/src/diffusers/pipelines/pipeline_utils.py#L562

2. Now you check if these kwargs are in `pipeline.config`. If they are you should retrieve them and then pass them

Reply:
actually, I don't quite understand this 

we are not calling the `__init__` method, we are calling the `from_pretrained()` - I think these steps are done inside `from_pretrained()` method no?